packages/manager/src/managers/slices/SlicesManager.ts

```diff
@@ -572,26 +572,37 @@ export class SlicesManager extends BaseManager {
 	async updateSliceModelScreenshotsInPlace(
 		args: SlicesManagerUpsertHostedSliceScrenshotsArgs,
 	): Promise<CustomTypes.Widgets.Slices.SharedSlice> {
-		const sliceMachineConfig = await this.project.getSliceMachineConfig();
-
-		const variations = await Promise.all(
+		const screenshots = await Promise.all(
 			args.model.variations.map(async (variation) => {
 				const screenshot = await this.readSliceScreenshot({
 					libraryID: args.libraryID,
 					sliceID: args.model.id,
 					variationID: variation.id,
 				});
 
+				return [screenshot.data, variation] as const;
+			}),
+		);
+
+		const hasSomeScreenshotChanged = screenshots.some(
+			([screenshot, variation]) =>
+				!screenshot ||
+				!variation.imageUrl.includes(createContentDigest(screenshot)),
+		);
+		if (hasSomeScreenshotChanged) {
+			await this.screenshots.deleteScreenshotFolder({ sliceID: args.model.id });
+		}
+
+		const sliceMachineConfig = await this.project.getSliceMachineConfig();
+		const variations = await Promise.all(
```

Comment:
If I get it right, this has the risk of deleting the folder, and then only uploading new screenshots, having unchanged screenshots being deleted as a result. Not sure this method should be updated, or we should only delete old screenshots that have changed individually in the next map.